### ISSUE TITLE
Support 2D barcode recognition

### ISSUE THREAD

**Issue by Vendo232** [blakeblackshear/frigate#3155]
Wyze just introduced a garage opener which registers door open/closed based on specific 2D barcode glued on the garage door.

![image](https://user-images.githubusercontent.com/44098230/165572977-f8c88e3d-5147-42a7-a394-7bf9d1f2e0ce.png)

Would something like this be possible with Frigate -> pushing status to HA?

**Comment by hawkeye217** [comment]
Agreed, @NickM-27. This seems outside of the use case for an NVR, but with Frigate's API, getting a snapshot and then passing it on with a script of some sort should be a fairly easy. It seems more suited to a HA custom component with ``image_processing`` (much like license plate recognition) rather than something in Frigate directly.

**Comment by hawkeye217** [comment]
Looks like there's a [Wyze custom integration](https://github.com/JoshuaMulliken/ha-wyzeapi) for Home Assistant already - you could probably see if they plan to implement something in the component for image processing. It would be easy to use a Frigate snapshot camera from Home Assistant at that point.

**Comment by Vendo232** [comment]
maybe it will show up in Wyze AP and then in HA

**Comment by Vendo232** [comment]
![image](https://user-images.githubusercontent.com/44098230/171975477-3230a475-efcc-46d7-8e7f-fc08731c718c.png)

**Comment by edk4971** [comment]
Here's another way to implement this feature that might work: https://hackaday.com/2022/07/08/use-apriltags-to-let-guests-open-your-front-gate/

**Comment by chrislawso** [comment]
@Vendo232 @edk4971 or anyone else has anyone gotten frigate and/or HA to scan 2d barcodes qrcodes or apriltags and can share your configuration in HA, yaml, etc of your setup? thanks

**Comment by 3h50** [comment]
> [@Vendo232](https://github.com/Vendo232) [@edk4971](https://github.com/edk4971) or anyone else has anyone gotten frigate and/or HA to scan 2d barcodes qrcodes or apriltags and can share your configuration in HA, yaml, etc of your setup? thanks

I am in the same boat... sometimes this will report numbers but not when im trying a qr code, just randomly in the logs.  There is little to 0 documentation

**Comment by grantgarbe** [comment]
I also thought of this idea and would find it useful to be implemented. I could make QR codes a key to unlock my front door with Home Assistant.
